aten/src/ATen/core/dispatch/DispatchKeyExtractor.h

```diff
@@ -55,6 +55,11 @@ namespace detail {
     void operator()(const at::Tensor& x) {
       ts = ts | x.key_set();
     }
+    void operator()(c10::optional<at::Tensor> x) {
+      if (x.has_value() && x->defined()) {
```

Comment:
Why does this need to check if it's defined but the above one doesn't?

Reply:
+1 @Chillee  here, the defined test is unnecessary, because (1) you aren't supposed to pass undefined tensors in and, (2) even if you pass it in, undefined tensors have defined key set which is just empty bitset.

Reply:
yup agree, will remove